src/mesh/parallel.jl

```diff
@@ -16,16 +19,38 @@ function partition!(mesh::ParallelTreeMesh)
   mesh.n_cells_by_rank = similar(n_leaves_per_rank)
 
   leaf_count = 0
-  last_id = leaves[n_leaves_per_rank[0]]
   mesh.first_cell_by_rank[0] = 1
-  mesh.n_cells_by_rank[0] = last_id
-  mesh.tree.mpi_ranks[1:last_id] .= 0
-  for d in 1:(length(n_leaves_per_rank)-1)
-    leaf_count += n_leaves_per_rank[d-1]
-    last_id = leaves[leaf_count + n_leaves_per_rank[d]]
-    mesh.first_cell_by_rank[d] = mesh.first_cell_by_rank[d-1] + mesh.n_cells_by_rank[d-1]
+  # Iterate over all ranks
+  for d in 0:(length(n_leaves_per_rank) - 1)
```

Comment:
I think this is easier to understand (if you change it, also fix it below)
```suggestion
  for d in 0:(mpi_nranks() - 1)
```

Reply:
You're right. That was already there before I changed it and I didn't think about it.